app/presenters/register_api/single_application_presenter.rb

```diff
@@ -0,0 +1,245 @@
+            english_language_qualifications: application_form.english_language_qualification_details,
+            other_languages: application_form.other_language_details,
+            disability_disclosure: application_form.disability_disclosure, # do we need this?
+            gender: application_form.equality_and_diversity.try(:[], 'sex'),
+            disabilities: application_form.equality_and_diversity.try(:[], 'disabilities'),
+            ethnic_group: application_form.equality_and_diversity.try(:[], 'ethnic_group'),
+            ethnic_background: application_form.equality_and_diversity.try(:[], 'ethnic_background'),
+          },
+          contact_details: contact_details,
+          course: course_info_for(application_choice.offered_option),
+          provider: provider_info_for(application_choice.offered_option.course.provider),
+          qualifications: qualifications,
+          hesa_itt_data: hesa_itt_data,
+        },
+      }
+    end
+
+  private
+
+    attr_reader :application_choice, :application_form
+
+    def status
+      application_choice.status
+    end
+
+    def uk_residency_status
+      return 'UK Citizen' if application_choice.nationalities.include?('GB')
+
+      return 'Irish Citizen' if application_choice.nationalities.include?('IE')
+
+      return application_form.right_to_work_or_study_details if application_form.right_to_work_or_study_yes?
+
+      'Candidate needs to apply for permission to work and study in the UK'
+    end
+
+    def uk_residency_status_code
+      return 'A' if application_choice.nationalities.include?('GB')
+      return 'B' if application_choice.nationalities.include?('IE')
+      return 'D' if application_form.right_to_work_or_study_yes?
+
+      'C'
+    end
+
+    def provisional_fee_payer_status
+      return UCAS_FEE_PAYER_CODES['SLC,SAAS,NIBd,EU,Chl,IoM'] if provisionally_eligible_for_gov_funding?
+
+      UCAS_FEE_PAYER_CODES['Not Known']
+    end
+
+    def provisionally_eligible_for_gov_funding?
+      return true if (PROVISIONALLY_ELIGIBLE_FOR_GOV_FUNDING_COUNTRY_CODES & application_choice.nationalities).any?
+
+      (EU_EEA_SWISS_COUNTRY_CODES & application_choice.nationalities).any? &&
+        application_form.right_to_work_or_study_yes? &&
+        application_form.uk?
+    end
+
+    def course_info_for(course_option)
+      {
+        recruitment_cycle_year: course_option.course.recruitment_cycle_year,
+        course_code: course_option.course.code,
+        course_name: course_option.course.name,
+        level: course_option.course.level,
+        subject_codes: course_option.course.subject_codes,
+        program_type: course_option.course.program_type,
+        start_date: course_option.course.start_date.strftime('%Y-%m'),
+        # Register would like end_date but we can't calculate it becuase
+        # course_length is a string. Typical values are 'OneYear', 'Two years'
+        # and some free text inputs
```

Comment:
should we return the `end_date` anyway even if it's a string, or would it be of no use in that format?

Reply:
- How would it be if we cut all fields that weren't necessary for identifying the course option? (ie leave only `recruitment_cycle_year`, `course_code`, `provider_code`, `site_code` and `study_mode`). Register can collect the rest, including `accredited_body`, from the TTAPI anyway. 
- Should we call the `provider_code` `training_provider_code` to be unambiguous?